src/GafferOSLModule/GafferOSLModule.cpp

```diff
@@ -140,10 +205,21 @@ BOOST_PYTHON_MODULE( _GafferOSL )
 	def( "oslLibraryVersionPatch", &oslLibraryVersionPatch );
 	def( "oslLibraryVersionCode", &oslLibraryVersionCode );
 
-	IECorePython::RefCountedClass<ShadingEngine, IECore::RefCounted>( "ShadingEngine" )
-		.def( init<const IECore::ObjectVector *>() )
-		.def( "shade", &ShadingEngine::shade, shadeOverloads() )
-	;
+
+	{
+		scope s = IECorePython::RefCountedClass<ShadingEngine, IECore::RefCounted>( "ShadingEngine" )
+			.def( init<const IECore::ObjectVector *>() )
+			.def( "shade", &shadeWrapper, shadeOverloads() )
```

Comment:
Rather than use `shadeOverloads()`, you can tell boost::python what the names and default values for the arguments are instead, and the python help string for the function will be more helpful. There's a decent example in the [Metadata bindings](https://github.com/GafferHQ/gaffer/blob/master/src/GafferBindings/MetadataBinding.cpp#L416).